core/util/notifications.py

```diff
@@ -135,6 +136,7 @@ def send_loan_expiry_message(
             f"Patron {loan.patron.authorization_identifier} has {len(tokens)} device tokens. "
             f"Sending loan expiry notification(s)."
         )
+        loan.patron_last_notified = utc_now().date()
```

Comment:
If firebase gives us a transient exception in `send_messages`, this might cause the notification to be dropped, instead of retried on the next script run. Since we set the notification date before we know that we sent a message.

Reply:
I've updated both instances of the code to only update the `patron_last_notified` if at least one notification was sent successfully.